### PR TITLE
Standalone migration follow-up fixes

### PR DESCRIPTION
**Pull request by crisbeto**
Addresses a couple of the follow-up discussion from the earlier standalone migration PRs. See the the individual commits for context.

### CODE REVIEW

File: packages/core/schematics/test/standalone_migration_spec.ts [hunk 1]
```diff
@@ -1413,6 +1413,61 @@ describe('standalone migration', () => {
...
   });
 
+  it('should not remove a chain of modules if a module in the chain cannot be removed',
+     async () => {
+       const moduleAContent = `
+          import {NgModule, InjectionToken} from '@angular/core';
+
+          export const token = new InjectionToken<any>('token');
+
+          @NgModule({providers: [{provide: token, useValue: 123}]})
```
Comment (devversion) [inline packages/core/schematics/test/standalone_migration_spec.ts:1423]: Optional if you follow-up on this anyway: Should we have a test for two more scenarios in the final module?

* Import to another *external* module, not using a property access. e.g. `SomeModuleWithProviders`
* import with a property access, which falls under your module with providers heuristic.

We already mentioned the partial evaluator for making this more reliable, but you mentioned that you would want to test some applications more to figure out if this is needed. there was also the case of normal identifiers that refer to e.g. `routes.ts` and an exported constant.

**Review by devversion — APPROVED** [review]

File: packages/core/schematics/test/standalone_migration_spec.ts (hunk 1 above)
Comment (devversion) [reply packages/core/schematics/test/standalone_migration_spec.ts:1423]: IIRC we had a migration doing that using the `PartialEvaluator` back in a while (by inspecting the type— we require explicit types for module with providers for that reason IIRC). But feel free to look into it as a follow-up/or acknowledge that this is not needed.

**Review by devversion — APPROVED** [review]

File: packages/core/schematics/test/standalone_migration_spec.ts (hunk 1 above)
Comment (crisbeto) [reply packages/core/schematics/test/standalone_migration_spec.ts:1423]: I'll leave this in the category of "see how common it is in real apps".

**Comment by jessicajaniuk** [comment]
This PR was merged into the repository by commit f82bdc4b01f93a7103870449d37da61cc4c4f179.

**Pull request closed without merging**